api/buffer/design.go

```diff
@@ -0,0 +1,549 @@
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			exportId()
+			Required("receiverId", "exportId")
+		})
+		HTTP(func() {
+			DELETE("/receivers/{receiverId}/exports/{exportId}")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+			ExportNotFoundError()
+		})
+	})
+
+	Method("Import", func() {
+		Meta("openapi:summary", "Import data")
+		Description("Upload data into the receiver. The data will be stored until import conditions are met.")
+		Payload(func() {
+			receiverId()
+			Attribute("secret", String, func() {
+				Description("Secret used for authentication.")
+				MinLength(48)
+				MaxLength(48)
+				Example("UBdJHwifkaQxbVwPyaRstdYpcboGwksSluCGIUWKttTiUdVH")
+			})
+			Attribute("data", Any, func() {
+				Meta(oneof.Meta, json.MustEncodeString([]*openapi.Schema{
+					{Type: openapi.String},
+					{Type: openapi.Integer},
+					{Type: openapi.Number},
+					{Type: openapi.Boolean},
+					{Type: openapi.Array, Items: &openapi.Schema{Type: openapi.String}},
+					{Type: openapi.Object},
+				}, false))
+				Example("foobar")
+			})
+			Required("data", "receiverId", "secret")
+		})
+		HTTP(func() {
+			POST("/import/{receiverId}/#/{secret}")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+			PayloadTooLargeError()
+			ResourceLimitReachedError()
+		})
+	})
+})
+
+// Common attributes
+
+var tokenSecurity = APIKeySecurity("storage-api-token", func() {
+	Description("Storage Api Token Authentication.")
+})
+
+var receiverId = func(desc ...string) {
+	Attribute("receiverId", String, func() {
+		if len(desc) > 0 {
+			Description(fmt.Sprintf("Unique ID of the receiver. %s", strings.Join(desc, "")))
+		} else {
+			Description("Unique ID of the receiver.")
+		}
+		MinLength(1)
+		MaxLength(48)
+	})
+}
+
+var exportId = func(desc ...string) {
+	Attribute("exportId", String, func() {
+		if len(desc) > 0 {
+			Description(fmt.Sprintf("Unique ID of the export. %s", strings.Join(desc, "")))
+		} else {
+			Description("Unique ID of the export.")
+		}
+		MinLength(1)
+		MaxLength(48)
+	})
+}
+
+var name = func(what string) {
+	Attribute("name", String, func() {
+		Description(fmt.Sprintf("Human readable name of the %s.", what))
+		MinLength(1)
+		MaxLength(40)
+	})
+}
+
+// Types --------------------------------------------------------------------------------------------------------------
+
+var ServiceDetail = Type("ServiceDetail", func() {
+	Description("Information about the service")
+	Attribute("api", String, "Name of the API", func() {
+		Example("buffer")
+	})
+	Attribute("documentation", String, "Url of the API documentation.", func() {
+		Example("https://buffer.keboola.com/v1/documentation")
+	})
+	Required("api", "documentation")
+})
+
+var Receiver = Type("Receiver", func() {
+	Description("Represents an endpoint for importing data. A project may have a maximum of 100 receivers.")
+	receiverId("May be null, in which case it will be generated.")
```

Comment:
`May be null` -> `May be omitted`?
Asi je beznejsie to neuvadzat vobec, ako nastavit na `null`, fungovat by mali oba pristupy.
(je to tu viac krat, komentujem to iba tu).

Reply:
A este by sa dalo napisat, ze `generated from the name`.

Reply:
+ `The value cannot be changed later.`